scylla/src/transport/metrics.rs

```diff
@@ -0,0 +1,92 @@
+use histogram::Histogram;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::{Arc, Mutex};
+
+const ORDER_TYPE: Ordering = Ordering::Relaxed;
+
+pub struct Metrics {
+    errors_num: AtomicU64,
+    queries_num: AtomicU64,
+    errors_iter_num: AtomicU64,
+    queries_iter_num: AtomicU64,
+    histogram: Arc<Mutex<Histogram>>,
+}
+
+impl Metrics {
+    pub fn new() -> Metrics {
+        Self {
+            errors_num: AtomicU64::new(0),
+            queries_num: AtomicU64::new(0),
+            errors_iter_num: AtomicU64::new(0),
+            queries_iter_num: AtomicU64::new(0),
+            histogram: Arc::new(Mutex::new(Histogram::new())),
+        }
+    }
+
+    /// Increments counter for errors that occured in single-page queries.
+    pub fn log_error(&self) {
+        self.errors_num.fetch_add(1, ORDER_TYPE);
+    }
+
+    /// Increments counter for single-page queries.
+    pub fn log_query(&self) {
+        self.queries_num.fetch_add(1, ORDER_TYPE);
+    }
+
+    /// Increments counter for errors that occured in iterative queries.
+    pub fn log_error_iter(&self) {
+        self.errors_iter_num.fetch_add(1, ORDER_TYPE);
+    }
+
+    /// Increments counter for page queries in iterative queries.
+    /// If query_iter would return 4 pages then this counter should be incremented 4 times.
+    pub fn log_query_iter(&self) {
+        self.queries_iter_num.fetch_add(1, ORDER_TYPE);
+    }
+
+    /// Saves to histogram latency of completing single query.
+    /// For iterative queries it should log latency for every page.
+    ///
+    /// # Arguments
+    ///
+    /// * `latency` - time in milliseconds that should be logged
+    pub fn log_finished_query(&self, latency: u64) {
```

Comment:
Here, `log_` makes more sense, if we add a little bit more context what we're logging: `log_query_latency`